miner/algo_common.go

```diff
@@ -281,15 +296,28 @@ func (envDiff *environmentDiff) commitBundle(bundle *types.SimulatedBundle, chDa
 			return errInterrupt
 		}
 
+		// update isRevertibleTx if transaction is found in list of reverting transaction hashes
+		isRevertibleTx = bundle.OriginalBundle.RevertingHash(txHash)
+
 		receipt, _, err := tmpEnvDiff.commitTx(tx, chData)
 
-		if err != nil {
-			log.Trace("Bundle tx error", "bundle", bundle.OriginalBundle.Hash, "tx", tx.Hash(), "err", err)
+		hasCommitErr = err != nil
+		hasReceiptFailed = receipt != nil && receipt.Status == types.ReceiptStatusFailed
+
+		// if drop enabled, and revertible tx has error on commit, we skip the transaction and continue with next one
+		if algoConf.DropRevertibleTxOnErr && isRevertibleTx && hasCommitErr {
```

Comment:
I think it'd be more readable to exhaust all possibilities for err/receipt/DropRevertibleTxOnErr/RevertingHash in a big switch or if-else. There's like four cases so it's not trivial any more

Reply:
updated